src/App.tsx

```diff
@@ -2,24 +1,57 @@
+import React, { useState } from 'react';
+
 import './App.css';
 import { getNumbers } from './utils';
+import { Pagination } from './components/Pagination';
 
-// eslint-disable-next-line @typescript-eslint/no-unused-vars
-const items = getNumbers(1, 42).map(n => `Item ${n}`);
+const total = 42;
+const startingPage = 5;
 
 export const App: React.FC = () => {
+  const [currentPage, setCurrentPage] = useState(1);
+  const [perPage, setPerPage] = useState(startingPage);
+
+  const items = getNumbers(1, total);
+  const totalPages = Math.ceil(items.length / perPage);
+
+  const indexOfLastItem = currentPage * perPage;
+  const indexOfFirstItem = indexOfLastItem - perPage;
+  const currentItems = items.slice(indexOfFirstItem, indexOfLastItem);
+
+  function handleSelectChange(event: React.ChangeEvent<HTMLSelectElement>) {
+    setPerPage(Number(event.target.value));
+    setCurrentPage(1);
+  }
+
+  function handleClick(page: number) {
+    setCurrentPage(page);
+  }
+
+  function showSubtitle(): string {
+    if (currentPage === totalPages) {
+      return `Page ${currentPage} (items ${indexOfFirstItem + 1} - ${items.length} of ${total})`;
+    }
+
+    return `Page ${currentPage} (items ${indexOfFirstItem + 1} - ${indexOfLastItem} of ${total})`;
+  }
+
   return (
     <div className="container">
       <h1>Items with Pagination</h1>
 
       <p className="lead" data-cy="info">
-        Page 1 (items 1 - 5 of 42)
+        {showSubtitle()}
```

Comment:
Using the 'showSubtitle' function directly in the JSX is fine, but if the subtitle computation is costly, consider memoizing the result with 'useMemo' to avoid unnecessary recalculations on each render.